src/cli/cli.cpp

```diff
@@ -1015,7 +1015,7 @@ void Interpreter::ProcessDiscover(int argc, char *argv[])
 
     if (argc > 0)
     {
-        SuccessOrExit(error = ParseLong(argv[0], value));
+        VerifyOrExit(ParseLong(argv[0], value) == OT_ERROR_NONE, error = OT_ERROR_INVALID_ARGS);
```

Comment:
What about reverting to the original `SuccessOrExit()` given that `ParseLong` will also return `OT_ERROR_INVALID_ARGS`?